src/test/java/com/wordCount/WordCounterUnitTests.java

```diff
@@ -0,0 +1,42 @@
+package com.wordCount;
+
+import com.wordCount.paramsSource.TestParamValuesSource;
+import com.wordcount.domain.WordCounter;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class WordCounterUnitTests {
+
+    @Test
+    public void word_counting_in_a_text_made_of_words_only_is_correct() {
+
+        HashMap<String, Integer> params = TestParamValuesSource.getWordsOnlyParamValues();
+
+        for (Map.Entry<String, Integer> param : params.entrySet()) {
+            checkThatWordsCountIsCorrect(param.getKey(), param.getValue().intValue());
+        }
+    }
+
+
+    @Test
+    public void word_counting_in_a_text_made_not_of_words_only_is_correct() {
+
+        HashMap<String, Integer> params = TestParamValuesSource.getNonWordsParamValues();
+
+        for (Map.Entry<String, Integer> param : params.entrySet()) {
+            checkThatWordsCountIsCorrect(param.getKey(), param.getValue().intValue());
+        }
+    }
+
+
+    private void checkThatWordsCountIsCorrect(String text, Integer wordCount) {
+        WordCounter sut = new WordCounter(text);
+        int wordNumber = sut.count();
+        assertEquals(wordCount, wordNumber);
+    }
```

Comment:
In general ok but with a little improved English, static imports, method references and renaming of variables you could improve readability quite a bit including a bit more conciseness.

```Java
@Test
public void count_words_in_given_text_which_contains_words() {
    paramsWithWords().forEach(this::assertUserInputTextWithExpectedWordCount);
}


@Test
public void count_words_in_given_text_which_contains_words_and_nonwords() {
    paramsWithWordsAndNonwords().forEach(this::assertUserInputTextWithExpectedWordCount);
}


private void assertUserInputTextWithExpectedWordCount(String userInputText, Integer expectedWordCount) {
    WordCounter sut = new WordCounter(userInputText);
    int actualWordCount = sut.count();
    assertEquals(expectedWordCount,
            actualWordCount,
            errorMessageFor(userInputText, expectedWordCount));
}

private String errorMessageFor(String userInputText, Integer expectedWordCount) {
    return format("user input text: '%s' is expected to contain '%d' words",
            userInputText,
            expectedWordCount);
}
```

Even though I don't mind reducing duplication in tests if readability is ok, I also would be ok by duplicating a bit to make tests self-contained. Tests should tell a story which means that you are allowed to have duplication for that reason.

```Java
@Test
public void count_words_in_given_text_which_contains_words() {
    paramsWithWords().forEach((userInputText, expectedWordCount) -> {
        WordCounter sut = new WordCounter(userInputText);
        int actualWordCount = sut.count();
        assertEquals(expectedWordCount,
                actualWordCount,
                errorMessageFor(userInputText, expectedWordCount));
    });
}


@Test
public void count_words_in_given_text_which_contains_words_and_nonwords() {
    paramsWithWordsAndNonwords().forEach((userInputText, expectedWordCount) -> {
        WordCounter sut = new WordCounter(userInputText);
        int actualWordCount = sut.count();
        assertEquals(expectedWordCount,
                actualWordCount,
                errorMessageFor(userInputText, expectedWordCount));
    });
}

private String errorMessageFor(String userInputText, Integer expectedWordCount) {
    return format("user input text: '%s' is expected to contain '%d' words",
            userInputText,
            expectedWordCount);
}
```